### PR TITLE
VFS-590 VFS-617 Proceed with moveTo operation if the exec channel for…

### PR DESCRIPTION
**Pull request by apearc03**
Pull request based on previous pr27 here.
https://github.com/apache/commons-vfs/pull/27

I've restructured the org.apache.commons.vfs2.provider.sftp.test package so that it contains an abstract test case similar to org.apache.commons.vfs2.provider.ftps.test.
The existing SFTP test cases are now split into individual implementations.

I've added a unit test to replicate the exec channel being closed for permission checks. An IOException is thrown during permission checks "id -G" or "id -u" meaning SftpFileSystem#executeCommand returns a failed exit status code.

### CODE REVIEW

File: commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderClosedExecChannelTestCase.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package org.apache.commons.vfs2.provider.sftp.test;
```
Comment (garydgregory) [inline commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderClosedExecChannelTestCase.java]: ASL header is missing causing the build to fail on Travis.

File: commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderStreamProxyModeTestCase.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
+package org.apache.commons.vfs2.provider.sftp.test;
+
```
Comment (garydgregory) [inline commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderStreamProxyModeTestCase.java]: ASL header is missing causing the build to fail on Travis.

**Comment by garydgregory** [comment]
@apearc03 
Thank you for your PR.
Please rebase on master and add the missing Apache License headers.
Thank you!

File: commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderStreamProxyModeTestCase.java (hunk 2 above)
Comment (apearc03) [reply commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderStreamProxyModeTestCase.java]: added

File: commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderClosedExecChannelTestCase.java (hunk 1 above)
Comment (apearc03) [reply commons-vfs2/src/test/java/org/apache/commons/vfs2/provider/sftp/test/SftpProviderClosedExecChannelTestCase.java]: added

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java [hunk 3]
```diff
@@ -76,12 +80,18 @@
...
+     */
+    private boolean execDisabled;
+
     protected SftpFileSystem(final GenericFileName rootName, final Session session,
             final FileSystemOptions fileSystemOptions) {
         super(rootName, null, fileSystemOptions);
         this.session = Objects.requireNonNull(session, "session");
         this.connectTimeoutMillis = SftpFileSystemConfigBuilder.getInstance()
                 .getConnectTimeoutMillis(fileSystemOptions);
+        detectExecDisabled();
```
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java]: @apearc03 

Thank you for updating the ASL headers.

Unless you expect the state of a server to change and re-detect `execDisabled`, please make `execDisabled` `final` and make `detectExecDisabled()` return a `boolean`.  A boolean method would be cleaner anyway IMO. Then the ctor can say 'this.execDisable = detectExecDisabled();`

We can always drop the `final` on the ivar later if `detectExecDisabled()` needs to be called post construction.

Comment (apearc03) [reply commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java]: Made changes to cleanup exec ```execDisabled``` initialization.

**Comment by LoicDelorme** [comment]
@garydgregory any new release incoming which includes this fix?

**Comment by garydgregory** [comment]
> @garydgregory any new release incoming which includes this fix?

I might take the time over the holidays (b/w now and the new year) to cut some Commons releases... no guarantees. There are a few Commons components that I can see releasing aside from this one as well. You can subscribe to our Commons Developer's mailing list to track progress, https://commons.apache.org/mail-lists.html

**Comment by simonalexander2005** [comment]
Hi, did this go anywhere? I'm having the same issue in 2.9.0

**Comment by garydgregory** [comment]
Hi @simonalexander2005 
The PR was mered, so you should try git master or a snapshot build from https://repository.apache.org/content/repositories/snapshots/

**Pull request merged**